cogs/autoemoji.py

```diff
@@ -0,0 +1,269 @@
+                update={"$set": {"emojis": emoji_list}},
+                upsert=True
+            )
+            self.autoemoji_list.append({"channel_id": interaction.channel.id, "emojis": emoji_list})
+        except Exception as e:
+            await message.edit(
+                embed=nextcord.Embed(
+                    title="AutoEmoji - Error",
+                    description=f"エラーが発生しました。\n```{e}```",
+                    color=self.bot.color.ERROR
+                )
+            )
+            return
+        await message.edit(
+            embed=nextcord.Embed(
+                title="AutoEmoji - Success",
+                description=(
+                    f"絵文字 {emojis} (`{emojis}`)をこのチャンネルに追加しました。\n"
+                    "このチャンネルにメッセージが送信されると自動で絵文字リアクションが行われます。"
+                ),
+                color=self.bot.color.NORMAL
+            )
+        )
+
+    @application_checks.guild_only()
+    @autoemoji_slash.subcommand(
+        name="del",
+        description="Delete autoemoji from this channel",
+        description_localizations={
+            nextcord.Locale.ja: "このチャンネルの自動絵文字を削除します。",
+        }
+    )
+    async def del_autoemoji_slash(self, interaction: Interaction):
+        await interaction.response.defer(ephemeral=False)
+        assert isinstance(interaction.guild, nextcord.Guild)
+        assert interaction.channel is not None
+        delete_status = await self.autoemoji_collection.delete_one({"channel_id": interaction.channel.id})
+        if delete_status.deleted_count == 0:
+            await interaction.send(
+                embed=nextcord.Embed(
+                    title="AutoEmoji - Error",
+                    description="このチャンネルには自動絵文字が設定されていません。",
+                    color=self.bot.color.ERROR
+                ),
+                ephemeral=True
+            )
+            return
+        await interaction.send(
+            embed=nextcord.Embed(
+                title="AutoEmoji - Success",
+                description=(
+                    "このチャンネルの自動絵文字を削除しました。\n"
+                    "このメッセージにつく絵文字が最後の絵文字だよ..."
+                ),
+                color=self.bot.color.NORMAL
+            ),
+            ephemeral=True
+        )
+        for autoemoji in self.autoemoji_list:
+            if autoemoji["channel_id"] == interaction.channel.id:
+                self.autoemoji_list.remove(autoemoji)
+                break
+
+    @application_checks.guild_only()
+    @autoemoji_slash.subcommand(
+        name="list",
+        description="List autoemoji settings",
+        description_localizations={
+            nextcord.Locale.ja: "自動絵文字の設定を表示します。",
+        }
+    )
+    async def list_autoemoji_slash(self, interaction: Interaction):
+        await interaction.response.defer(ephemeral=False)
+        assert interaction.channel is not None
+
+        autoemoji_settings = list(filter(lambda x: x["channel_id"] == interaction.channel.id, self.autoemoji_list))
```

Comment:
静的チェックでは lambda がいつ実行されるかわからないから正しく検証できないよ  
https://github.com/microsoft/pyright/issues/2343
```suggestion
        assert interaction.channel is not None
        channel = interaction.channel

        autoemoji_settings = list(filter(lambda x: x["channel_id"] == channel.id, self.autoemoji_list))
```